app/models/user.rb

```diff
@@ -1002,6 +1007,33 @@ def from_staged?
     custom_fields[User::FROM_STAGED]
   end
 
+  def totp
```

Comment:
Our `User` class is already quite big. At the very least I think we might want to put this in a mixin and include it.

We could also consider creating a helper class like `UserTotp.new(user)` then call `create`, `provision_url`, etc on it.